src/main/java/com/epam/page/object/generator/util/PropertyLoader.java

```diff
@@ -26,29 +28,41 @@ public SearchRuleGroupsScheme getMapWithScheme() {
         Map<String, Schema> schemeMap = new HashMap<>();
 
         logger.info("Start reading list of schemes...");
-        JSONObject jsonObject = new JSONObject(
-            new JSONTokener(PropertyLoader.class.getResourceAsStream(propertyFile)));
-
-        JSONArray typeGroups = jsonObject.getJSONArray("typeGroups");
-        for (int i = 0; i < typeGroups.length(); i++) {
-
-            JSONObject group = typeGroups.getJSONObject(i);
-            String groupName = group.getString("name");
-            logger.debug("Read '" + groupName + "' group...");
-
-            String schemaPath = group.getString("schema");
-
-            try {
-                JSONObject jsonSchema = new JSONObject(
-                    new JSONTokener(
-                        PropertyLoader.class.getResourceAsStream(schemaPath)));
-                Schema schema = SchemaLoader.load(jsonSchema);
-                schemeMap.put(groupName, schema);
-                logger.debug("Add schema = '" + schemaPath + "' for '" + groupName + "' group");
-            } catch (NullPointerException e){
-                logger.error("Schema = '" + schemaPath + "' doesn't exist!", e);
-                throw new NullPointerException("\nSchema = '" + schemaPath + "' doesn't exist!");
+        try (InputStream propertyStream = PropertyLoader.class.getResourceAsStream(propertyFile)) {
+            JSONObject jsonObject = new JSONObject(
+                new JSONTokener(propertyStream));
+
+            JSONArray typeGroups = jsonObject.getJSONArray("typeGroups");
+            for (int i = 0; i < typeGroups.length(); i++) {
+
+                JSONObject group = typeGroups.getJSONObject(i);
+                String groupName = group.getString("name");
+                logger.debug("Read '" + groupName + "' group...");
+
+                String schemaPath = group.getString("schema");
+
+                try (InputStream schemaStream = PropertyLoader.class
+                    .getResourceAsStream(schemaPath)) {
+                    JSONObject jsonSchema = new JSONObject(
+                        new JSONTokener(schemaStream));
+                    Schema schema = SchemaLoader.load(jsonSchema);
+                    schemeMap.put(groupName, schema);
+                    logger.debug("Add schema = '" + schemaPath + "' for '" + groupName + "' group");
+                } catch (NullPointerException e) {
```

Comment:
If exception was caught here, it is thrown as nullpointer exception with correct message on line 53.
On line 61 correct message would be caught andrethrown as "Property file doesn't exist" even though real problem is absence of schema file.